### PR TITLE
Add new param "subMenuIcon" at AddSubMenu RPC

### PR DESCRIPTION
**Pull request by VVeremjova**
Added in AddSubMenu new parameter subMenuIcon
Related to [APPLINK-23379](https://adc.luxoft.com/jira/browse/APPLINK-23379)

### CODE REVIEW

**Comment by VVeremjova** [comment]
Please review @TMelnyk, @AByzhynar, @dcherniev, @AGritsevich, @Kozoriz, @dtrunov, @pvvasilev

File: src/components/application_manager/src/commands/mobile/add_sub_menu_request.cc [hunk 1]
```diff
@@ -150,6 +167,41 @@ bool AddSubMenuRequest::CheckSubMenuName() {
...
+  full_file_path += file_system::GetPathDelimiter();
+  full_file_path +=
+      (*message_)[strings::msg_params][strings::sub_menu_icon][strings::value]
+          .asString();
+
+  if (!file_system::FileExists(full_file_path)) {
+    return false;
+  }
+  return false;
+}
```
Comment (dtrunov) [inline src/components/application_manager/src/commands/mobile/add_sub_menu_request.cc]: @VVeremjova May be you need to write here return true instead return false.

File: src/components/application_manager/src/hmi_capabilities_impl.cc [hunk 2]
```diff
@@ -257,6 +257,9 @@ void InitCapabilities() {
       std::string("menuIcon"), hmi_apis::Common_ImageFieldName::menuIcon));
   image_field_name_enum.insert(std::make_pair(
       std::string("cmdIcon"), hmi_apis::Common_ImageFieldName::cmdIcon));
+  image_field_name_enum.insert(
```
Comment (dtrunov) [inline src/components/application_manager/src/hmi_capabilities_impl.cc]: @VVeremjova Why did you add this to hmi_capabilities?. This map using for convertation string parameter from hmi_capabilities.json to enum. I didn't see that you add some parameters to hmi_capabilities.json file.

File: src/components/application_manager/src/message_helper.cc [hunk 3]
```diff
@@ -1344,6 +1344,13 @@ smart_objects::SmartObjectList MessageHelper::CreateAddSubMenuRequestToHMI(
     msg_params[strings::menu_id] = i->first;
     msg_params[strings::menu_params][strings::position] =
         (*i->second)[strings::position];
+
```
Comment (dtrunov) [inline src/components/application_manager/src/message_helper.cc]: @VVeremjova  As far as I understand you need to add supporting saving this new parameter to resumption DB. Please clarify this with Melnyk T.

File: src/components/application_manager/src/smart_object_keys.cc [hunk 4]
```diff
@@ -339,6 +340,7 @@ const char* show_strings = "showStrings";
 const char* interaction_layout = "interactionLayout";
 const char* menu_title = "menuTitle";
 const char* menu_icon = "menuIcon";
+const char* sub_menu_icon = "subMenuIcon";
```
Comment (dtrunov) [inline src/components/application_manager/src/smart_object_keys.cc]: @VVeremjova Please remove one subMenuIcon key.

File: src/components/application_manager/src/hmi_capabilities_impl.cc (hunk 2 above)
Comment (VVeremjova) [inline src/components/application_manager/src/hmi_capabilities_impl.cc]: @dtrunov Fixed in 133961d

File: src/components/application_manager/src/commands/mobile/add_sub_menu_request.cc (hunk 1 above)
Comment (VVeremjova) [inline src/components/application_manager/src/commands/mobile/add_sub_menu_request.cc]: @dtrunov Fixed in 133961d

File: src/components/application_manager/src/commands/mobile/add_sub_menu_request.cc [hunk 5]
```diff
@@ -150,6 +167,41 @@ bool AddSubMenuRequest::CheckSubMenuName() {
   return true;
 }
 
+bool AddSubMenuRequest::CheckSubMenuIcon() {
+  if (!(*message_)[strings::msg_params][strings::sub_menu_icon].keyExists(
+          strings::value)) {
```
Comment (VVeremjova) [inline src/components/application_manager/src/commands/mobile/add_sub_menu_request.cc]: @dtrunov Fixed in 133961d

File: src/components/application_manager/include/application_manager/smart_object_keys.h [hunk 6]
```diff
@@ -97,6 +97,7 @@ extern const char* cmd_id;
 extern const char* menu_params;
 extern const char* menu_title;
 extern const char* menu_icon;
+extern const char* sub_menu_icon;
 extern const char* keyboard_properties;
```
Comment (VVeremjova) [inline src/components/application_manager/include/application_manager/smart_object_keys.h]: @dtrunov Fixed in 133961d

**Comment by VVeremjova** [comment]
Please review @TMelnyk, @AByzhynar, @dcherniev, @AGritsevich, @Kozoriz, @dtrunov, @pvvasilev

File: src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc [hunk 7]
```diff
@@ -219,6 +264,123 @@ TEST_F(AddSubMenuRequestTest, Run_SUCCESS) {
...
+          (*result_msg)[am::strings::params][am::strings::function_id].asInt());
+  EXPECT_EQ(hmi_apis::FunctionID::UI_AddSubMenu, kReceivedResult);
+
+  EXPECT_EQ("icon.png",
+            (*result_msg)[am::strings::msg_params][am::strings::sub_menu_icon]
+                         [am::strings::value].asString());
+}
+
+TEST_F(AddSubMenuRequestTest, Run_MenuIconNewLineChar_SendWithoutIcon) {
+  MessageSharedPtr command_msg(CreateMessage(smart_objects::SmartType_Map));
```
Comment (Kozoriz) [inline src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc]: @VVeremjova , please create this in fixture

File: src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc [hunk 8]
```diff
@@ -219,6 +264,123 @@ TEST_F(AddSubMenuRequestTest, Run_SUCCESS) {
...
+  (*command_msg)[am::strings::params][am::strings::connection_key] =
+      kConnectionKey;
+  (*command_msg)[am::strings::msg_params][am::strings::menu_name] =
+      "valid_sub_menu_name";
+  (*command_msg)[am::strings::msg_params][am::strings::sub_menu_icon]
+                [am::strings::value] = "ico n.png";
+
+  AddSubMenuPtr command(CreateCommand<AddSubMenuRequest>(command_msg));
+
+  MockAppPtr app(CreateMockApp());
```
Comment (Kozoriz) [inline src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc]: @VVeremjova , please move app to fixture.

File: src/components/application_manager/test/include/application_manager/test_resumption_data_db.h [hunk 9]
```diff
@@ -53,7 +55,11 @@ class TestResumptionDataDB : public ResumptionDataDB {
   application_manager_test::MockApplicationManagerSettings
       mock_application_manager_settings_;
   TestResumptionDataDB(DbStorage db_storage)
-      : ResumptionDataDB(db_storage, mock_application_manager_settings_) {}
+      : ResumptionDataDB(db_storage, mock_application_manager_settings_) {
+    //      const std::string storage_folder = "./";
+    //      ON_CALL(mock_application_manager_settings_, app_storage_folder()).
```
Comment (Kozoriz) [inline src/components/application_manager/test/include/application_manager/test_resumption_data_db.h]: @VVeremjova , why this commented?

File: src/components/application_manager/test/resumption/resumption_data_db_test.cc [hunk 10]
```diff
@@ -824,7 +831,7 @@ TEST_F(ResumptionDBTest_WithMockStorage,
 }
 
 TEST_F(ResumptionDBTest_WithMockStorage,
-       SaveApplication_AppNotChanged_AppExist_DBProblem) {
+       DISABLED_SaveApplication_AppNotChanged_AppExist_DBProblem) {
```
Comment (Kozoriz) [inline src/components/application_manager/test/resumption/resumption_data_db_test.cc]: @VVeremjova , why this disabled. PLease fix, or create task and todo to fix.

File: src/components/application_manager/test/resumption/resumption_data_test.cc [hunk 11]
```diff
@@ -129,9 +129,22 @@ void ResumptionDataTest::CheckSubmenues(sm::SmartObject& res_list) {
     std::string name =
         (*test_submenu_map[i + 10])[am::strings::menu_name].asString();
     int position = (*test_submenu_map[i + 10])[am::strings::position].asInt();
+    std::string icon_name =
```
Comment (Kozoriz) [inline src/components/application_manager/test/resumption/resumption_data_test.cc]: @VVeremjova , what is the magic number 10?

File: src/components/application_manager/test/resumption/resumption_data_test.cc [hunk 12]
```diff
@@ -465,6 +478,9 @@ void ResumptionDataTest::SetSubmenues() {
     sm_comm[am::strings::menu_id] = i;
     sm_comm[am::strings::position] = i;
     sm_comm[am::strings::menu_name] = "SubMenu" + std::string(numb);
+    sm_comm[am::strings::sub_menu_icon][am::strings::value] =
+        "submenuicon" + std::string(numb);
```
Comment (Kozoriz) [inline src/components/application_manager/test/resumption/resumption_data_test.cc]: @VVeremjova , can you move string(numb) creating before SO filling?

File: src/components/application_manager/test/resumption/resumption_sql_queries_test.cc [hunk 13]
```diff
@@ -2170,7 +2172,7 @@ TEST_F(ResumptionSqlQueriesTest,
 TEST_F(ResumptionSqlQueriesTest, kInsertToSubMenu_ExpectDataInserted) {
   // Arrange
   SQLQuery temp_query(db());
-  FillSubMenuTable(temp_query, 1, "tst_menu", 3);
+  FillSubMenuTable(temp_query, 1, "tst_menu", 3, 4);
```
Comment (Kozoriz) [inline src/components/application_manager/test/resumption/resumption_sql_queries_test.cc]: @VVeremjova , a lot of magic numbers .

File: src/components/application_manager/test/resumption/resumption_sql_queries_test.cc [hunk 14]
```diff
@@ -1380,7 +1383,7 @@ TEST_F(ResumptionSqlQueriesTest, kDeleteSubMenu_ExpectDataDeleted) {
 }
 
 TEST_F(ResumptionSqlQueriesTest,
-       kDeleteApplicationSubMenuArray_ExpectDataDeleted) {
+       DISABLED_kDeleteApplicationSubMenuArray_ExpectDataDeleted) {
```
Comment (Kozoriz) [inline src/components/application_manager/test/resumption/resumption_sql_queries_test.cc]: @VVeremjova , one more disabled test.

File: src/components/application_manager/src/commands/mobile/add_sub_menu_request.cc [hunk 15]
```diff
@@ -93,6 +94,16 @@ void AddSubMenuRequest::Run() {
       (*message_)[strings::msg_params][strings::menu_name];
   msg_params[strings::app_id] = app->app_id();
 
+  if (((*message_)[strings::msg_params].keyExists(strings::sub_menu_icon)) &&
+      CheckSubMenuIcon()) {
+    msg_params[strings::sub_menu_icon] = (*message_)[strings::sub_menu_icon];
```
Comment (VVeremjova) [inline src/components/application_manager/src/commands/mobile/add_sub_menu_request.cc]: @dtrunov See 	3d5b88d

**Comment by TMelnyk** [comment]
auto-commit review

File: src/components/application_manager/test/resumption/resumption_sql_queries_test.cc (hunk 13 above)
Comment (VVeremjova) [inline src/components/application_manager/test/resumption/resumption_sql_queries_test.cc]: @Kozoriz See 775092b and 3d5b88d

File: src/components/application_manager/test/resumption/resumption_sql_queries_test.cc (hunk 14 above)
Comment (VVeremjova) [inline src/components/application_manager/test/resumption/resumption_sql_queries_test.cc]: @Kozoriz See 775092b and 3d5b88d

File: src/components/application_manager/test/resumption/resumption_data_test.cc (hunk 11 above)
Comment (VVeremjova) [inline src/components/application_manager/test/resumption/resumption_data_test.cc]: @Kozoriz See 775092b and 3d5b88d

File: src/components/application_manager/test/resumption/resumption_data_db_test.cc (hunk 10 above)
Comment (VVeremjova) [inline src/components/application_manager/test/resumption/resumption_data_db_test.cc]: @Kozoriz See 775092b and 3d5b88d

File: src/components/application_manager/test/include/application_manager/test_resumption_data_db.h (hunk 9 above)
Comment (VVeremjova) [inline src/components/application_manager/test/include/application_manager/test_resumption_data_db.h]: @Kozoriz See 775092b and 3d5b88d

File: src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc (hunk 8 above)
Comment (VVeremjova) [inline src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc]: @Kozoriz See 3d5b88d

File: src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc (hunk 7 above)
Comment (VVeremjova) [inline src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc]: @Kozoriz See 3d5b88d

File: src/components/application_manager/src/commands/mobile/add_sub_menu_request.cc [hunk 16]
```diff
@@ -150,6 +164,38 @@ bool AddSubMenuRequest::CheckSubMenuName() {
   return true;
 }
 
+bool AddSubMenuRequest::CheckSubMenuIcon() {
+  if (!(*message_)[strings::msg_params].keyExists(strings::sub_menu_icon)) {
+    return false;
+  }
+  std::string str =
```
Comment (AByzhynar) [inline src/components/application_manager/src/commands/mobile/add_sub_menu_request.cc]: @VVeremjova Why it isn't `const`?

File: src/components/application_manager/src/commands/mobile/add_sub_menu_request.cc [hunk 17]
```diff
@@ -150,6 +164,38 @@ bool AddSubMenuRequest::CheckSubMenuName() {
...
+      (*message_)[strings::msg_params][strings::sub_menu_icon][strings::value]
+          .asString();
+  if (!CheckSyntax(str) || std::string::npos != str.find(" ")) {
+    SDL_ERROR("Invalid menu icon syntax check failed.");
+    return false;
+  }
+
+  return true;
+}
+bool AddSubMenuRequest::CheckMenuIconExistedInStorage() {
```
Comment (AByzhynar) [inline src/components/application_manager/src/commands/mobile/add_sub_menu_request.cc]: @VVeremjova Please separate different methods definitions with an empty line

File: src/components/application_manager/src/message_helper.cc [hunk 18]
```diff
@@ -1344,6 +1344,13 @@ smart_objects::SmartObjectList MessageHelper::CreateAddSubMenuRequestToHMI(
...
+    if (((*i->second).keyExists(strings::sub_menu_icon)) &&
+        (0 < (*i->second)[strings::sub_menu_icon][strings::value].length())) {
```
Comment (AByzhynar) [inline src/components/application_manager/src/message_helper.cc]: @VVeremjova Use `!empty()` instead of comparison of length with `0`

File: src/components/application_manager/src/message_helper.cc [hunk 19]
```diff
@@ -1344,6 +1344,13 @@ smart_objects::SmartObjectList MessageHelper::CreateAddSubMenuRequestToHMI(
     msg_params[strings::menu_id] = i->first;
     msg_params[strings::menu_params][strings::position] =
         (*i->second)[strings::position];
+
+    if (((*i->second).keyExists(strings::sub_menu_icon)) &&
```
Comment (AByzhynar) [inline src/components/application_manager/src/message_helper.cc]: @VVeremjova Can we create an alias for (*i->second) at the beginning of loop and use it instead?

File: src/components/application_manager/test/CMakeLists.txt [hunk 20]
```diff
@@ -70,8 +70,8 @@ set(RESUMPTION_PATHS
 )
 # Temporary exclude some resumption tests from build due to compile errors APPLINK-26346
 set(RESUMPTION_EXCLUDE
-  resumption_data_json_test.cc
-  resumption_data_db_test.cc
+#  resumption_data_json_test.cc
+#  resumption_data_db_test.cc
```
Comment (AByzhynar) [inline src/components/application_manager/test/CMakeLists.txt]: @VVeremjova Why did you comment tests above?

File: src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc [hunk 21]
```diff
@@ -70,7 +70,15 @@ const int32_t kMenuId = 5;
 }  // namespace
 
 class AddSubMenuRequestTest
-    : public CommandRequestTest<CommandsTestMocks::kIsNice> {};
+    : public CommandRequestTest<CommandsTestMocks::kIsNice> {
+ public:
```
Comment (AByzhynar) [inline src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc]: @VVeremjova `public` -> `protected`

File: src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc [hunk 22]
```diff
@@ -109,6 +114,49 @@ TEST_F(AddSubMenuRequestTest, OnEvent_SUCCESS) {
   EXPECT_EQ(mobile_apis::Result::SUCCESS, kReceivedResult);
 }
 
+TEST_F(AddSubMenuRequestTest,
+       OnEvent_SubMenuWithIcon_NotExistsInFolder_ExpectInfo) {
+  // message will be send on hmi with icon that not exists in folder
```
Comment (AByzhynar) [inline src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc]: @VVeremjova Comments should start from capital letter

File: src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc [hunk 23]
```diff
@@ -109,6 +114,49 @@ TEST_F(AddSubMenuRequestTest, OnEvent_SUCCESS) {
...
+
+  (*command_msg_)[am::strings::msg_params][am::strings::menu_id] = kMenuId;
+  (*command_msg_)[am::strings::params][am::strings::connection_key] =
+      kConnectionKey;
+  (*command_msg_)[am::strings::msg_params][am::strings::sub_menu_icon]
+                 [am::strings::value] = "icon.png";
+
+  Event event(hmi_apis::FunctionID::UI_AddSubMenu);
```
Comment (AByzhynar) [inline src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc]: @VVeremjova Please move variables to the place before their first usage. In all test cases.

File: src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc [hunk 24]
```diff
@@ -109,6 +114,49 @@ TEST_F(AddSubMenuRequestTest, OnEvent_SUCCESS) {
...
+      CatchMobileCommandResult(CallOnEvent(*command, event)));
+  const mobile_apis::Result::eType kReceivedResult =
+      static_cast<mobile_apis::Result::eType>(
+          (*result_msg)[am::strings::msg_params][am::strings::result_code]
+              .asInt());
+
+  EXPECT_EQ(mobile_apis::Result::WARNINGS, kReceivedResult);
+  const std::string received_info =
+      (*result_msg)[am::strings::msg_params][am::strings::info].asString();
+  EXPECT_EQ("Reference image(s) not found", received_info);
```
Comment (AByzhynar) [inline src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc]: @VVeremjova 1) Don't compare `const char*` to `string` even that it will be implicitly converted.
2) Please create const string for "Reference image(s) not found"

File: src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc [hunk 25]
```diff
@@ -217,6 +282,86 @@ TEST_F(AddSubMenuRequestTest, Run_SUCCESS) {
...
+      "valid_sub_menu_name";
+  (*command_msg_)[am::strings::msg_params][am::strings::sub_menu_icon]
+                 [am::strings::value] = "ico\nn.png";
+
+  AddSubMenuPtr command(CreateCommand<AddSubMenuRequest>(command_msg_));
+
+  EXPECT_CALL(mock_app_manager_, application(kConnectionKey))
+      .WillOnce(Return(app));
+  EXPECT_CALL(*app, FindSubMenu(kMenuId))
+      .WillOnce(Return(static_cast<SmartObject*>(NULL)));
```
Comment (AByzhynar) [inline src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc]: @VVeremjova Can we avoid casting?

File: src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc [hunk 26]
```diff
@@ -195,18 +235,43 @@ TEST_F(AddSubMenuRequestTest, Run_NotValidSubMenuName_UNSUCCESS) {
...
+
+  MessageSharedPtr result_msg(CatchHMICommandResult(CallRun(*command)));
+  const hmi_apis::FunctionID::eType kReceivedResult =
+      static_cast<hmi_apis::FunctionID::eType>(
+          (*result_msg)[am::strings::params][am::strings::function_id].asInt());
+  EXPECT_EQ(hmi_apis::FunctionID::UI_AddSubMenu, kReceivedResult);
+}
+
+TEST_F(AddSubMenuRequestTest, Run_MenuIconCorrectName_SUCCESS) {
+  (*command_msg_)[am::strings::msg_params][am::strings::menu_id] = kMenuId;
```
Comment (dtrunov) [inline src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc]: @VVeremjova I see same line of code about filling message in all test cases may be move this code to separate method.

File: src/components/interfaces/HMI_API.xml [hunk 27]
```diff
@@ -2851,6 +2854,11 @@
     <param name="appID" type="Integer" mandatory="true">
       <description>ID of application that requested this RPC.</description>
     </param>
+    <param name="subMenuIcon" type="Common.Image" mandatory="false">
+    <description>
+      If ommited on supported displays, no (or the default if applicable) icon shall be displayed.
```
Comment (AByzhynar) [inline src/components/interfaces/HMI_API.xml]: @VVeremjova `ommited` is wrong. Correct one is `omitted`. In all descriptions

File: src/components/application_manager/test/resumption/resumption_sql_queries_test.cc [hunk 28]
```diff
@@ -2633,9 +2644,9 @@ TEST_F(ResumptionSqlQueriesTest, kSelectSubMenu_ExpectDataCorrect) {
                                          device_id,
                                          9,
                                          true).LastInsertId();
+  int64_t key1 = FillImageTable(temp_query, 1, test_image).LastInsertId();
```
Comment (AByzhynar) [inline src/components/application_manager/test/resumption/resumption_sql_queries_test.cc]: @VVeremjova  Can be const?

File: src/components/application_manager/test/resumption/resumption_sql_queries_test.cc [hunk 29]
```diff
@@ -234,7 +235,8 @@ class ResumptionSqlQueriesTest : public ::testing::Test {
   SQLQuery& FillSubMenuTable(SQLQuery& query,
                              const int menuID,
                              const string& menuName,
-                             const int position);
+                             const int position,
+                             const int64_t idimage);
```
Comment (AByzhynar) [inline src/components/application_manager/test/resumption/resumption_sql_queries_test.cc]: @VVeremjova What does it mean `idimage`? If it is id for image - rename variable to `image_id`

File: src/components/application_manager/test/resumption/resumption_data_test.cc [hunk 30]
```diff
@@ -459,12 +475,17 @@ void ResumptionDataTest::SetCommands() {
 
 void ResumptionDataTest::SetSubmenues() {
   sm::SmartObject sm_comm;
```
Comment (AByzhynar) [inline src/components/application_manager/test/resumption/resumption_data_test.cc]: @VVeremjova Unclear name for me :`sm_comm`

File: src/components/application_manager/test/resumption/resumption_data_db_test.cc [hunk 31]
```diff
@@ -839,6 +846,8 @@ TEST_F(ResumptionDBTest_WithMockStorage,
   EXPECT_CALL(*app_ptr, is_application_data_changed())
       .WillRepeatedly(Return(false));
   EXPECT_CALL(*app_ptr, hmi_level()).WillRepeatedly(Return(HMILevel::HMI_FULL));
+  EXPECT_CALL(*app_ptr, IsAppSubscribedForWayPoints(_))
```
Comment (AByzhynar) [inline src/components/application_manager/test/resumption/resumption_data_db_test.cc]: @VVeremjova What about parameter check?

File: src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc (hunk 25 above)
Comment (VVeremjova) [inline src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc]: @AByzhynar Unfortunately, we cannot avoid it

File: src/components/application_manager/test/resumption/resumption_data_test.cc (hunk 30 above)
Comment (VVeremjova) [inline src/components/application_manager/test/resumption/resumption_data_test.cc]: @AByzhynar See f3d57ba

File: src/components/application_manager/test/resumption/resumption_sql_queries_test.cc (hunk 28 above)
Comment (VVeremjova) [inline src/components/application_manager/test/resumption/resumption_sql_queries_test.cc]: @AByzhynar See f3d57ba

File: src/components/application_manager/test/resumption/resumption_sql_queries_test.cc (hunk 29 above)
Comment (VVeremjova) [inline src/components/application_manager/test/resumption/resumption_sql_queries_test.cc]: @AByzhynar See f3d57ba

File: src/components/application_manager/test/resumption/resumption_data_db_test.cc [hunk 32]
```diff
@@ -374,9 +374,16 @@ void ResumptionDataDBTest::CheckSubmenuData() {
     std::string name =
         (*test_submenu_map[i])[am::strings::menu_name].asString();
     int position = (*test_submenu_map[i])[am::strings::position].asInt();
+    std::string image_value =
+        (*test_submenu_map[i])[am::strings::sub_menu_icon][am::strings::value]
+            .asString();
+    int image_type = (*test_submenu_map[i])[am::strings::sub_menu_icon]
```
Comment (VVeremjova) [inline src/components/application_manager/test/resumption/resumption_data_db_test.cc]: @AByzhynar See f3d57ba

File: src/components/application_manager/test/resumption/resumption_data_db_test.cc [hunk 33]
```diff
@@ -374,9 +374,16 @@ void ResumptionDataDBTest::CheckSubmenuData() {
     std::string name =
         (*test_submenu_map[i])[am::strings::menu_name].asString();
     int position = (*test_submenu_map[i])[am::strings::position].asInt();
+    std::string image_value =
```
Comment (VVeremjova) [inline src/components/application_manager/test/resumption/resumption_data_db_test.cc]: @AByzhynar See f3d57ba

File: src/components/application_manager/test/resumption/resumption_data_db_test.cc [hunk 34]
```diff
@@ -374,9 +374,16 @@ void ResumptionDataDBTest::CheckSubmenuData() {
     std::string name =
         (*test_submenu_map[i])[am::strings::menu_name].asString();
     int position = (*test_submenu_map[i])[am::strings::position].asInt();
```
Comment (VVeremjova) [inline src/components/application_manager/test/resumption/resumption_data_db_test.cc]: @AByzhynar See f3d57ba

File: src/components/application_manager/test/resumption/resumption_data_db_test.cc [hunk 35]
```diff
@@ -374,9 +374,16 @@ void ResumptionDataDBTest::CheckSubmenuData() {
     std::string name =
```
Comment (VVeremjova) [inline src/components/application_manager/test/resumption/resumption_data_db_test.cc]: @AByzhynar See f3d57ba

File: src/components/interfaces/HMI_API.xml (hunk 27 above)
Comment (VVeremjova) [inline src/components/interfaces/HMI_API.xml]: @AByzhynar  	9be9eb2

File: src/components/application_manager/src/commands/mobile/add_sub_menu_request.cc (hunk 16 above)
Comment (VVeremjova) [inline src/components/application_manager/src/commands/mobile/add_sub_menu_request.cc]: @AByzhynar See 9be9eb2

File: src/components/application_manager/src/commands/mobile/add_sub_menu_request.cc (hunk 17 above)
Comment (VVeremjova) [inline src/components/application_manager/src/commands/mobile/add_sub_menu_request.cc]: @AByzhynar See 9be9eb2

File: src/components/application_manager/src/message_helper.cc (hunk 19 above)
Comment (VVeremjova) [inline src/components/application_manager/src/message_helper.cc]: @AByzhynar See 9be9eb2

File: src/components/application_manager/src/message_helper.cc (hunk 18 above)
Comment (VVeremjova) [inline src/components/application_manager/src/message_helper.cc]: @AByzhynar See 9be9eb2

File: src/components/application_manager/test/CMakeLists.txt (hunk 20 above)
Comment (VVeremjova) [inline src/components/application_manager/test/CMakeLists.txt]: @AByzhynar See f3d57ba

File: src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc (hunk 21 above)
Comment (VVeremjova) [inline src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc]: @AByzhynar See f3d57ba

File: src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc (hunk 22 above)
Comment (VVeremjova) [inline src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc]: @AByzhynar See f3d57ba

File: src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc (hunk 23 above)
Comment (VVeremjova) [inline src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc]: @AByzhynar See f3d57ba

File: src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc (hunk 24 above)
Comment (VVeremjova) [inline src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc]: @AByzhynar See f3d57ba

**Comment by VVeremjova** [comment]
Closed due to moving CRQ to develop
